File: frontend/autocare-frontend/src/services/contactMessageService.js
import api from "./api";

const contactMessageService = {

    getAllMessages() {

        return api.get("/contact-messages");

    },

    getMessageById(id) {

        return api.get(`/contact-messages/${id}`);

    },

    createMessage(message) {

        return api.post("/contact-messages", message);

    },

    deleteMessage(id) {

        return api.delete(`/contact-messages/${id}`);

    }

};

export default contactMessageService;

File: autocare-backend/src/main/java/com/pobitra/autocare/service/ContactMessageService.java
package com.pobitra.autocare.service;

import com.pobitra.autocare.dto.ContactMessageRequestDTO;
import com.pobitra.autocare.dto.ContactMessageResponseDTO;

import java.util.List;

public interface ContactMessageService {

    ContactMessageResponseDTO createMessage(ContactMessageRequestDTO dto);

    List<ContactMessageResponseDTO> getAllMessages();

    ContactMessageResponseDTO getMessageById(Long id);

    List<ContactMessageResponseDTO> getMessagesByEmail(String email);

    void deleteMessage(Long id);

}